tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/RightToWorkValidator.java

```diff
@@ -90,35 +90,52 @@ private void checkSettled(RightToWorkDTO dto, List<FieldError> fieldErrors) {
   }
 
   private void checkVisaDates(List<FieldError> fieldErrors, RightToWorkDTO dto, Long personId) {
+
+    boolean dbValidationError = false;
+
     if (dto != null) {
       LocalDate visaIssued = dto.getVisaIssued();
       LocalDate visaValidTo = dto.getVisaValidTo();
-      Optional<Person> originalPersonRecord = personRepository.findPersonById(personId);
 
-      if (visaIssued != null && visaValidTo != null && visaIssued.isAfter(visaValidTo)) {
-        FieldError fieldError =
-            new FieldError(DTO_NAME, FIELD_NAME_VISA_ISSUED, "visaIssued must be before "
-                + "visaValidTo.");
-        fieldErrors.add(fieldError);
-      } else if (originalPersonRecord.isPresent()) {
-        Person existingPerson = originalPersonRecord.get();
-        RightToWork oldRtwDto = existingPerson.getRightToWork();
-        if (visaIssued != null && visaValidTo == null) {
-          if (visaIssued.isAfter(oldRtwDto.getVisaValidTo())) {
-            FieldError fieldError =
-                new FieldError(DTO_NAME, FIELD_NAME_VISA_ISSUED, "visaIssued is after "
-                    + "current visaValidTo date.");
-            fieldErrors.add(fieldError);
-          }
-        } else if (visaValidTo != null && visaIssued == null
-            && visaValidTo.isBefore(oldRtwDto.getVisaIssued())) {
+      if (visaIssued == null && visaValidTo == null) {
+        //Nothing to check
+        return;
+      }
+
+      if (visaIssued != null && visaValidTo != null) {
+        //only check the values passed in
+        if (visaIssued.isAfter(visaValidTo)) {
           FieldError fieldError =
-              new FieldError(DTO_NAME, FIELD_NAME_VISA_ISSUED, "visaValidTo date is "
-                  + "before current visaIssued date.");
+              new FieldError(DTO_NAME, FIELD_NAME_VISA_ISSUED, "visaIssued must be"
+                  + " before visaValidTo.");
           fieldErrors.add(fieldError);
         }
+      } else {
+        dbValidationError = checkDbVisaDates(personId, visaIssued, visaValidTo);
+      }
+
+      if (dbValidationError) {
+        FieldError fieldError =
+            new FieldError(DTO_NAME, FIELD_NAME_VISA_ISSUED, "Visa Dates conflict"
+                + " with dates already in Database");
```

Comment:
The error might be with the `visaValidTo` passed in rather than `visaIssued`.

You could inline the method if it doesn't break cognitive complexity constraints.  If it does you could 1)still inline the checksuse DB values to set whichever was null of `visaValidTo` or `visaIssued` and then do the compare, or 2)pass in `fieldErrors` to `checkDbVisaDates()`.